src/workerd/api/r2-bucket.c++

```diff
@@ -521,11 +523,22 @@ R2Bucket::put(jsg::Lock& js, kj::String name, kj::Maybe<R2PutValue> value,
     }
 
     auto requestJson = json.encode(requestBuilder);
+    auto account = adminAccount.map([](auto&& s) { return kj::str(s); });
     auto bucket = adminBucket.map([](auto&& s) { return kj::str(s); });
 
+    kj::Vector<kj::String> path;
+    KJ_IF_MAYBE(a, account) {
+      path.add(kj::mv(*a));
```

Comment:
This is incorrect and will cause the injection to only work once. You need to copy it since the binding may be used multiple times. Indicates that you may also be missing a test case here (i.e. do 2 operations on the R2Bucket instance)

Reply:
Just pushed some changes, please review that my new implementation is correct 🙏